### PR TITLE
Add recalculate msg after reconnected

### PR DESCRIPTION
**Pull request by AlanJager**
for https://github.com/zstackio/issues/issues/2808

### CODE REVIEW

File: plugin/nfsPrimaryStorage/src/main/java/org/zstack/storage/primary/nfs/NfsPrimaryStorage.java [hunk 1]
```diff
@@ -1185,6 +1185,10 @@ protected void done() {
                 done(new FlowDoneHandler(completion) {
                     @Override
                     public void handle(Map data) {
+                        RecalculatePrimaryStorageCapacityMsg rmsg = new RecalculatePrimaryStorageCapacityMsg();
```
Comment (zxwing) [inline plugin/nfsPrimaryStorage/src/main/java/org/zstack/storage/primary/nfs/NfsPrimaryStorage.java]: move to parent class

File: storage/src/main/java/org/zstack/storage/primary/PrimaryStorageCapacityUpdater.java [hunk 2]
```diff
@@ -115,6 +115,7 @@ private boolean isResized() {
         return originalCopy != null &&
                 capacityVO != null &&
                 originalCopy.getTotalPhysicalCapacity() != 0 &&
+                capacityVO.getTotalPhysicalCapacity() != 0 &&
```
Comment (zxwing) [inline storage/src/main/java/org/zstack/storage/primary/PrimaryStorageCapacityUpdater.java]: 去掉这个

File: test/src/test/groovy/org/zstack/test/integration/storage/primary/nfs/capacity/NfsCapacityCase.groovy [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        reconnectPrimaryStorage {
+            uuid = ps.uuid
+        }
+
+        GetPrimaryStorageCapacityResult afterReconnectPS = getPrimaryStorageCapacity {
+            primaryStorageUuids = [ps.uuid]
+        }
+        assert afterReconnectPS.availableCapacity == afterCreateVmInstance.availableCapacity
+        assert afterReconnectPS.totalCapacity == afterCreateVmInstance.totalCapacity
+    }
```
Comment (Alvin-Lau) [inline test/src/test/groovy/org/zstack/test/integration/storage/primary/nfs/capacity/NfsCapacityCase.groovy]: 建议增加 beforeCapacity.totalCapacity == afterReconnectPS.totalCapacity 的断言。

File: test/src/test/groovy/org/zstack/test/integration/storage/primary/smp/SMPCapacityCase.groovy [hunk 4]
```diff
@@ -84,6 +94,52 @@ class SMPCapacityCase extends SubCase{
...
+        assert afterReconnectPS.totalCapacity == afterCreateVmInstance.totalCapacity
+
+        destroyVmInstance {
+            uuid = vm.uuid
+        }
+
+        expungeVmInstance {
+            uuid = vm.uuid
+        }
+
```
Comment (Alvin-Lau) [inline test/src/test/groovy/org/zstack/test/integration/storage/primary/smp/SMPCapacityCase.groovy]: 建议删除云主机后，再次重连主存储。再次检查容量是否准确

Comment (AlanJager) [inline test/src/test/groovy/org/zstack/test/integration/storage/primary/smp/SMPCapacityCase.groovy]: fixed

**Pull request merged**